### ISSUE TITLE
unit test fails when component under test imoports a <Link> component from Gatsby

### ISSUE THREAD

**Issue by thescientist13** [thegreenhouseio/www.thegreenhouse.io#34]
So if you enable the <Header> component specs, this error will show in the terminal
```
 FAIL  src/components/header/header.spec.jsx
  Header Component
    ✕ should not be null (33ms)

  ● Header Component › should not be null

    TypeError: Cannot read property 'history' of undefined

      10 |
      11 |   beforeEach(() => {
    > 12 |     header = mount(<Header/>);
      13 |   });
      14 |
      15 |   // TODO fix why bombs out because of <Link> being used in the component

      at new GatsbyLink (node_modules/gatsby-link/index.js:107:34)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:292:18
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:73:12)
```

This is the same for any other component, and seems to have to do something with Gatsby providing some sort of functionality under the hood in the app?

Anyway, solving this would be ideal since it is preventing the app from having better code coverage.

**Comment by thescientist13** [comment]
This is line 107, so seems like some sort of Routing is expected?
```
    var history = context.router.history;
```

**Comment by thescientist13** [comment]
commented on an open issue with Gastby + Jest [here](https://github.com/gatsbyjs/gatsby/issues/2932#issuecomment-376306319)

**Comment by thescientist13** [comment]
resolved as part of #82 / #85
